src/Behaviors/Meshes/baseSixDofDragBehavior.ts

```diff
@@ -0,0 +1,378 @@
+import { Mesh } from "../../Meshes/mesh";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Scene } from "../../scene";
+import { Nullable } from "../../types";
+import { PointerInfo, PointerEventTypes } from "../../Events/pointerEvents";
+import { Vector3, Quaternion } from "../../Maths/math.vector";
+import { Observer, Observable } from "../../Misc/observable";
+import { TransformNode } from "../../Meshes/transformNode";
+import { Camera } from "../../Cameras/camera";
+
+/**
+ * Base behavior for six degrees of freedom interactions in XR experiences.
+ * Creates virtual meshes that are dragged around
+ * And observables for position/rotation changes
+ */
+export class BaseSixDofDragBehavior implements Behavior<Mesh> {
+    protected static _virtualScene: Scene;
+    private _pointerObserver: Nullable<Observer<PointerInfo>>;
+    private _attachedToElement: boolean = false;
+    protected _virtualMeshesInfo: {
+        [id: number]: {
+            dragging: boolean;
+            moving: boolean;
+            dragMesh: AbstractMesh;
+            originMesh: AbstractMesh;
+            pivotMesh: AbstractMesh;
+            startingPivotPosition: Vector3;
+            startingPivotOrientation: Quaternion;
+            startingPosition: Vector3;
+            startingOrientation: Quaternion;
+            lastOriginPosition: Vector3;
+            lastDragPosition: Vector3;
+        };
+    } = {};
+
+    private _tmpVector: Vector3 = new Vector3();
+    private _tmpQuaternion: Quaternion = new Quaternion();
+
+    protected _scene: Scene;
+    protected _moving = false;
+    protected _ownerNode: TransformNode;
+    protected _dragging: boolean = false;
+
+    /**
+     * The list of child meshes that can receive drag events
+     * If `null`, all child meshes will receive drag event
+     */
+    public draggableMeshes: Nullable<AbstractMesh[]> = null;
+
+    /**
+     * How much faster the object should move when the controller is moving towards it. This is useful to bring objects that are far away from the user to them faster. Set this to 0 to avoid any speed increase. (Default: 3)
+     */
+    public zDragFactor = 3;
+    /**
+     * The id of the pointer that is currently interacting with the behavior (-1 when no pointer is active)
+     */
+    public get currentDraggingPointerId() {
+        if (this.currentDraggingPointerIds[0] !== undefined) {
+            return this.currentDraggingPointerIds[0];
+        }
+        return -1;
+    }
+
+    public set currentDraggingPointerId(value: number) {
+        this.currentDraggingPointerIds[0] = value;
+    }
+
+    /**
+     * In case of multipointer interaction, all pointer ids currently active are stored here
+     */
+    public currentDraggingPointerIds: number[] = [];
+
+    /**
+     * Get or set the currentDraggingPointerId
+     * @deprecated Please use currentDraggingPointerId instead
+     */
+    public get currentDraggingPointerID(): number {

```

Comment:
Do we need to add those? no need to support an API we didn't have until now. Unless I am missing something :-)

Reply:
This was in `SixDofDragBehavior.ts`, so I kept it for backwards compatibility